File: day02/star03.py
with open('input.txt') as f:
    lines = f.readlines()

lines = [line.split() for line in lines]

# A/X = rock
# B/Y = paper
# C/Z = scissors

game_score = {'A': {'X': 3, 'Y': 6, 'Z': 0},
              'B': {'X': 0, 'Y': 3, 'Z': 6},
              'C': {'X': 6, 'Y': 0, 'Z': 3}}

play_score = {'X': 1, 'Y': 2, 'Z': 3}

total = 0
for opp, yours in lines:
    score = game_score[opp][yours] + play_score[yours]
    total += score

print(f"{total = }")

File: day02/star03.cpp
#include <fstream>
#include <string>
#include <iostream>
#include <map>

int main()
{
    std::map<std::string, int> game_score {{"A X", 3}, {"A Y", 6}, {"A Z", 0},
                                           {"B X", 0}, {"B Y", 3}, {"B Z", 6},
                                           {"C X", 6}, {"C Y", 0}, {"C Z", 3}};

    std::map<char, int> play_score {{'X', 1}, {'Y', 2}, {'Z', 3}};

    int total = 0;
    std::ifstream f {"input.txt"};
    std::string line;
    while (std::getline(f, line)) {
        total += game_score[line] + play_score[line[2]];
    }

    std::cout << "total = " << total << std::endl;
}
